### PR TITLE
chore(ci): Update check-spelling to 0.0.22

### PR DESCRIPTION
**Pull request by jszwedko**
## Summary

Updates the check-spelling action to 0.0.22

## Change Type
- [ ] Bug fix
- [ ] New feature
- [x] Non-functional (chore, refactoring, docs)
- [ ] Performance

## Is this a breaking change?
- [ ] Yes
- [x] No

## How did you test this PR?

Replaced the trigger with `pull_request` so that it would run the workflow version from this PR. Successful run: https://github.com/vectordotdev/vector/actions/runs/11730011340/job/32676900007

## Does this PR include user facing changes?

- [ ] Yes. Please add a changelog fragment based on our [guidelines](https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md).
- [x] No. A maintainer will apply the "no-changelog" label to this PR.

## Checklist
- [x] Please read our [Vector contributor resources](https://github.com/vectordotdev/vector/tree/master/docs#getting-started).
- [x] If this PR introduces changes Vector dependencies (modifies `Cargo.lock`), please
  run `dd-rust-license-tool write` to regenerate the [license inventory](https://github.com/vectordotdev/vrl/blob/main/LICENSE-3rdparty.csv) and commit the changes (if any). More details [here](https://crates.io/crates/dd-rust-license-tool).

## References

Reverts https://github.com/vectordotdev/vector/pull/21732

### CODE REVIEW

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -87,9 +87,9 @@ jobs:
     steps:
     - name: check-spelling
       id: spelling
-      uses: check-spelling/check-spelling@v0.0.21
+      uses: check-spelling/check-spelling@v0.0.22
       with:
-        suppress_push_for_open_pull_request: 1
+        suppress_push_for_open_pull_request: ${{ github.actor != 'dependabot[bot]' && 1 }}
```
Comment (jszwedko) [inline .github/workflows/spelling.yml:92]: Suggested by https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22

File: .github/workflows/spelling.yml [hunk 2]
```diff
@@ -98,34 +98,39 @@ jobs:
         use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
         extra_dictionary_limit: 20
         extra_dictionaries:
```
Comment (jszwedko) [inline .github/workflows/spelling.yml:100]: v0.0.22 reorganized the cspell dictionaries signficantly as mentioned on https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22 . I updated the list by removing all of them and looking at the suggestions from the `check-spell` action.

File: .github/actions/spelling/expect.txt [hunk 3]
```diff
@@ -1,5 +1,6 @@
 abcd
```
Comment (jszwedko) [inline .github/actions/spelling/expect.txt:1]: cspell dictionaries were changed so I think this resulted in more words needing to be added to the `expect.txt`. Updated by running the `apply.pl` suggestion from the `check-spelling` results.

File: website/cue/reference/remap/expressions/path.cue [hunk 4]
```diff
@@ -61,9 +61,9 @@ remap: expressions: path: {
 							"""
 					}
 					nonexistent: {
-						title: "Non-existent paths"
+						title: "Nonexistent paths"
```
Comment (jszwedko) [inline website/cue/reference/remap/expressions/path.cue:64]: Flagged by [forbidden patterns](https://github.com/vectordotdev/vector/blob/master/.github/actions/spelling/line_forbidden.patterns).

**Comment by jszwedko** [comment]
`pull_request_target` will fail since I updated `expect.txt` in this PR to be compatible with the new version of check-spelling

**Review by pront — APPROVED** [review]
Thank you!

**Comment by jszwedko** [comment]
Merged manually since the required changes to `expect.txt` were blocking merge when used with 0.0.22 (the version on master).

**Pull request merged**